puppet-checks/src/main/java/com/iadams/sonarqube/puppet/checks/CollapsibleIfStatementsCheck.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SonarQube Puppet Plugin
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Iain Adams and David RACODON
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.iadams.sonarqube.puppet.checks;
+
+import com.iadams.sonarqube.puppet.PuppetCheckVisitor;
+import com.iadams.sonarqube.puppet.api.PuppetGrammar;
+import com.iadams.sonarqube.puppet.api.PuppetTokenType;
+import com.sonar.sslr.api.AstNode;
+import java.util.List;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+import org.sonar.sslr.ast.AstSelect;
+
+@Rule(
+  key = "S1066",
+  priority = Priority.MAJOR,
+  name = "Collapsible \"if\" statements should be merged",
+  tags = {Tags.CONFUSING})
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
+@SqaleConstantRemediation("5min")
+public class CollapsibleIfStatementsCheck extends PuppetCheckVisitor {
+
+  @Override
+  public void init() {
+    subscribeTo(PuppetGrammar.IF_STMT, PuppetGrammar.ELSEIF_STMT);
+  }
+
+  @Override
+  public void visitNode(AstNode node) {
+    AstNode singleIfChild = singleIfChild(node);
+    if(singleIfChild != null && !hasElseOrElsif(singleIfChild)){
+      addIssue(singleIfChild, this, "Merge this if statement with the enclosing one.");
+    }
+  }
+
+  private boolean hasElseOrElsif(AstNode ifNode){
+    return ifNode.hasDirectChildren(PuppetGrammar.ELSEIF_STMT) || ifNode.hasDirectChildren(PuppetGrammar.ELSE_STMT);
```

Comment:
Shorter that way: `return ifNode.hasDirectChildren(PuppetGrammar.ELSEIF_STMT, PuppetGrammar.ELSE_STMT);`